lib/ach_client/providers/sftp/ach_transaction.rb

```diff
@@ -26,13 +26,30 @@ def to_entry_detail
       end
 
       private
+
+      def amount_in_cents
+        # Take absolute value in case amount is negative
+        Helpers::DollarsToCents.dollars_to_cents(amount.abs)
+      end
+
       def transaction_code
         [
           AccountTypeTransformer.serialize_to_provider_value(
             account_type
           ),
           TransactionTypeTransformer.serialize_to_provider_value(
-            transaction_type
+            # Some NACHA providers can't handle negative amounts
+            # So we flip the amount and toggle the transaction type between
+            #   Debit and Credit
+            if amount.negative?
```

Comment:
@phereford 
Agree this code chunk feels longer than it needs to be.
We need to flip the transaction type regardless of its original value. I'm not sure your suggestion would do that.
Can you think of anything else that would flip the transaction type iff the amount is negative?